src/micro_http/src/common/headers.rs

```diff
@@ -286,6 +298,72 @@ impl Headers {
     }
 }
 
+/// Wrapper over supported AcceptEncoding.
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub struct Encoding {}
+
+impl Encoding {
+    /// Parses a byte slice into a Encoding structure for a HTTP request. Encoding
+    /// must be ASCII, so also UTF-8 valid.
+    ///
+    /// # Errors
+    /// The function returns `InvalidRequest` when the byte stream is empty, `InvalidValue` when the
+    /// byte stream contains a logical error, `UnsupportedValue` when the byte stream contains a
+    /// value that is not supported yet and `InvalidUtf8String` when the byte stream contains
+    /// invalid characters.
```

Comment:
It's better that you start the explanation for each error variant with the error variant name: `ErrorOne is returned when scenario one happens`. Each possible error and its explanation should be on their own line. Additionally, you should leave an empty line between each error variant so that when formatted by cargo-doc it is displayed like a list. I know this is not how other comments in the crate are written, but it is a good practice nonetheless.